server/src/main/java/org/elasticsearch/action/bulk/BulkRequestParser.java

```diff
@@ -179,6 +185,17 @@ public void parse(
                     );
                 }
                 String action = parser.currentName();
+                if (SUPPORTED_ACTIONS.contains(action) == false) {
+                    deprecationLogger.compatibleCritical(
+                        STRICT_ACTION_PARSING_WARNING_KEY,
+                        String.format(
+                            Locale.ROOT,
+                            "Unsupported action: `%s`. Supported values are `create`, `delete`, `index`, and `update`. "
+                                + "Unsupported actions are currently accepted but will be rejected in a future version.",
+                            action
+                        )
+                    );
```

Comment:
`DeprecationLogger#compatibleCritical` looks to accept a format string and args already, do we need to use `String.format` here or could we just do this?

```suggestion
                    deprecationLogger.compatibleCritical(
                        STRICT_ACTION_PARSING_WARNING_KEY,
                            "Unsupported action: [{}]. Supported values are [create], [delete], [index], and [update]. "
                                + "Unsupported actions are currently accepted but will be rejected in a future version.",
                            action
                    );
```

Also I'd prefer `[brackets]` over backticks in these messages.